internal/schema/schema.go

```diff
@@ -511,10 +512,21 @@ func parseDirectiveDef(l *common.Lexer) *types.DirectiveDefinition {
 		l.ConsumeToken(')')
 	}
 
-	l.ConsumeKeyword("on")
+	switch x := l.ConsumeIdent(); x {
+	case "on":
+		// no-op; Go doesn't fallthrough by default
+	case "repeatable":
+		d.Repeatable = true
+		l.ConsumeKeyword("on")
+	default:
+		l.SyntaxError(fmt.Sprintf(`unexpected %q, expecting "on" or "repeatable"`, x))
+	}
 
 	for {
 		loc := l.ConsumeIdent()
+		if ok, err := regexp.MatchString("^[A-Z_]+$", loc); err != nil || !ok {
+			l.SyntaxError(fmt.Sprintf("expected directive location-spec to be SNAKE_CASE, but got %q", loc))
+		}
```

Comment:
Can this start with `_`? Also, why do we need the regex at all?

Reply:
Hrrrm, yeah, it would accept `_WHATEVER` (or just `_`, for that matter). Fixed.

The two obvious approaches, to me, are "regex-match" or "enumerate legal directive locations" -- I leaned towards regex-matching so that this library won't need updating if/when the GraphQL spec adds new legal locations in future revisions. But if you'd prefer the enumeration approach, I could re-instate [this commit](https://github.com/graph-gophers/graphql-go/pull/502/commits/10dda978ed415d5017ad6a5a41b032a8c77a937f) which I ended up reverting.

Reply:
I would prefer the white list, please. It is much stricter and gives a clear error message. The regex is not preferred. What is more in your example, the regex is not pre-compiled and it is going to be compiled on every directive.

Reply:
All righty, sounds good! Done.